streaming/text/convert/enwiki.py

```diff
@@ -0,0 +1,114 @@
+from argparse import ArgumentParser, Namespace
+import os
+from typing import Dict, List
+
+from tqdm import tqdm
+
+from streaming.base import MDSWriter
+from streaming.base.util import get_list_arg
+
+
+def parse_args() -> Namespace:
+    """Parse command-line arguments.
+
+    Args:
+        Namespace: command-line arguments.
+    """
+    args = ArgumentParser()
+    args.add_argument(
+        '--in_root',
+        type=str,
+        required=True,
+        help='Directory path to store the input dataset',
+    )
+    args.add_argument(
+        '--out_root',
+        type=str,
+        required=True,
+        help='Directory path to store the output dataset',
+    )
+    args.add_argument(
+        '--compression',
+        type=str,
+        default='zstd:7',
+        help='Compression algorithm to use. Default: zstd:7',
+    )
+    args.add_argument(
+        '--hashes',
+        type=str,
+        default='sha1,xxh64',
+        help='Hashing algorithms to apply to shard files. Default: sha1,xxh64',
+    )
+    args.add_argument(
+        '--size_limit',
+        type=int,
+        default=1 << 25,
+        help='Shard size limit, after which point to start a new shard. Default: 1 << 27',
+    )
+    args.add_argument(
+        '--progbar',
+        type=int,
+        default=1,
+        help='tqdm progress bar. Default: 1 (True)',
+    )
+    args.add_argument(
+        '--leave',
+        type=int,
+        default=0,
+        help='Keeps all traces of the progressbar upon termination of iteration. Default: 0 ' +
+        '(False)',
+    )
+    return args.parse_args()
+
+
+def process_split(in_root: str, out_root: str, compression: str, hashes: List[str], size_limit: int,
+                  progbar: int, leave: int, basenames: List[str], split: str) -> None:
+    """Process a dataset split.
```

Comment:
What does this mean? Process a dataset that has already been split to different shards?

Reply:
No. The `process_split()` reads the raw dataset (for example, which we downloaded from internet) and converts into `*.mds` file extension with multiple shard files.

Reply:
@knighton maybe it needs a better descriptive method comments ?

Reply:
In this case basenames is already a collection of 500 txt files.

Reply:
Got it. So it's just reading the files one-by-one but sharding etc. is handled by the writer.